File: engine/platform/graphics/ImageFilter.cpp
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "config.h"
#include "ImageFilter.h"

#include "third_party/skia/include/core/SkImageFilter.h"

namespace blink {

FloatRect mapImageFilterRect(ImageFilter* filter, const FloatRect& rect)
{
    SkRect dest;
    filter->computeFastBounds(rect, &dest);
    return dest;
}

} // namespace blink

File: lib/web_ui/lib/src/engine/compositor/image_filter.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

part of engine;

/// The CanvasKit implementation of [ui.ImageFilter].
///
/// Currently only supports `blur`.
class SkImageFilter implements ui.ImageFilter {
  js.JsObject skImageFilter;

  SkImageFilter.blur({double sigmaX = 0.0, double sigmaY = 0.0}) {
    skImageFilter = canvasKit['SkImageFilter'].callMethod(
      'MakeBlur',
      <dynamic>[
        sigmaX,
        sigmaY,
        canvasKit['TileMode']['Clamp'],
        null,
      ],
    );
  }
}
